### DISCUSSION TITLE
Accessing the RACK via Fuseki?

### DISCUSSION THREAD

**Discussion by Robert-Adelard** [ge-high-assurance/RACK#456] [Q&A] [answered]
I notice that version 7.0 of the RACK now exports port 3030, making Fuseki accessible. I've tried using this interface to explore the contents of the RACK, but  I can't see any datasets...

1. What is the rationale for exposing Fuseki?
2. How do I use the Fuseki interface?

Thanks.

**Comment by cuddihyge** [comment]
@tuxji might be able to explain why 3030 is exposed.

Fuseki is opensource and documented [here](https://jena.apache.org/documentation/fuseki2/)

It is possible the shiro.ini is set up to disallow non-administrators to edit datasets.

You'll have to be more specific about what you're trying to do and why.

**Reply by tuxji** [reply]
Technically, we only changed the "docker run" command in our README and Wiki documentation to add a "-p 3030:3030" option which tells Docker to expose Fuseki's port when it runs the container.  The rack-box image itself didn't need any changes.  I didn't make that change to the documentation; Galois submitted a PR changing that documentation when they added a tool of theirs called "analyze" to the rack-box image.  I guess they wanted to let people run "analyze" locally on their host and have it talk directly to Fuseki's 3030 port.

**Reply by Robert-Adelard** [reply]
Thanks - I'd be interested to know why Galois felt it was necessary to bypass the SemTk layer and talk to Fuseki directly.

**Reply by kquick** [reply]
We opened up the direct Fuseki port to allow performers to run the various assist tools against their Docker image (see https://github.com/ge-high-assurance/RACK/tree/master/assist#readme and https://github.com/ge-high-assurance/RACK/tree/master/assist/bin#readme).  As described in those documents, assist can be used to perform validation of the data loaded into RACK, and also supports the prototype support for build process data capture and upload.

Turnstile data is supplemented by the latter process above and will look more complete when both stages are performed.

The assist tooling uses Prolog-based logic reasoning to perform the functionality; Prolog has libraries supporting direct RDF-database interaction, which is why this does not use the SemTK interface.

**Comment by Robert-Adelard** [comment, marked as answer]
I was trying to view the RACK dataset, but there were no datasets visible...

> It is possible the shiro.ini is set up to disallow non-administrators to
> edit datasets.

Not quite - according to the Fuseki documentation:

> In its default configuration, SPARQL endpoints are open to the public but administrative functions are limited to localhost. 


https://jena.apache.org/documentation/fuseki2/fuseki-security.html

So the Fuseki interface on port 3030 is only usable if the RACK is running on localhost. Otherwise, no datasets are visible.
